--- minisat-220/minisat/core/Solver.h	2013-09-25 14:16:18.000000000 +0200
+++ minisat-220-mod/minisat/core/Solver.h	2015-01-08 14:59:59.000000000 +0100
@@ -37,6 +37,13 @@
 class Solver {
 public:
 
+    void* termCallbackState;
+    int (*termCallback)(void* state);
+    void setTermCallback(void* state, int (*termCallback)(void*)) {
+	this->termCallbackState = state;
+	this->termCallback = termCallback;
+    }
+
     // Constructor/Destructor:
     //
     Solver();
@@ -372,7 +379,7 @@
 inline void     Solver::clearInterrupt(){ asynch_interrupt = false; }
 inline void     Solver::budgetOff(){ conflict_budget = propagation_budget = -1; }
 inline bool     Solver::withinBudget() const {
-    return !asynch_interrupt &&
+    return !asynch_interrupt && (termCallback == NULL || 0 == termCallback(termCallbackState)) &&
            (conflict_budget    < 0 || conflicts < (uint64_t)conflict_budget) &&
            (propagation_budget < 0 || propagations < (uint64_t)propagation_budget); }
 
